### PR TITLE
std.experimental.ndslice: improve docs

### PR DESCRIPTION
**Pull request by wilzbach**
I had a quick pass through ndslice's docs - I should start to do this every week ;-)
Ping @9il

### CODE REVIEW

File: std/experimental/ndslice/selection.d [hunk 1]
```diff
@@ -27,12 +27,12 @@ $(T2 evertPack, reverses dimension packs)
 $(BOOKTABLE $(H2 Selectors),
 
 $(TR $(TH Function Name) $(TH Description))
-$(T2 byElement, a random access range of all elements with `index` property)
+$(T2 byElement, flat range of all elements with `index` property)
```
Comment (9il) [inline std/experimental/ndslice/selection.d]: `random access` make sense

File: std/experimental/ndslice/selection.d [hunk 2]
```diff
@@ -27,12 +27,12 @@ $(T2 evertPack, reverses dimension packs)
...
 $(T2 byElementInStandardSimplex, an input range of all elements in standard simplex of hypercube with `index` property.
     If the slice has two dimensions, it is a range of all elements of upper left triangular matrix.)
-$(T2 indexSlice, returns a lazy slice with elements equal to the initial multidimensional index)
-$(T2 iotaSlice, returns a lazy slice with elements equal to the initial flattened (continuous) index)
-$(T2 reshape, returns a new slice for the same data)
+$(T2 indexSlice, new slice with initial multidimensional index)
```
Comment (9il) [inline std/experimental/ndslice/selection.d]: `new` refers to something allocated. `indexSlice` does not allocate anything

File: std/experimental/ndslice/selection.d [hunk 3]
```diff
@@ -27,12 +27,12 @@ $(T2 evertPack, reverses dimension packs)
...
+$(T2 iotaSlice, new slice with initial flattened (continuous) index)
```
Comment (9il) [inline std/experimental/ndslice/selection.d]: ditto

File: std/experimental/ndslice/selection.d (hunk 2 above)
Comment (9il) [inline std/experimental/ndslice/selection.d]: so we need note that this is lazy

File: std/experimental/ndslice/selection.d [hunk 4]
```diff
@@ -27,12 +27,12 @@ $(T2 evertPack, reverses dimension packs)
...
+$(T2 reshape, new slice with changed dimensions)
```
Comment (9il) [inline std/experimental/ndslice/selection.d]: ditto, need note that data is the same

File: std/experimental/ndslice/slice.d [hunk 5]
```diff
@@ -1700,7 +1700,7 @@ struct Slice(size_t _N, _Range)
     }
 
     /++
-    $(BLUE Fully defined index).
+    Fully defined index
```
Comment (9il) [inline std/experimental/ndslice/slice.d]: Bold please

File: std/experimental/ndslice/slice.d [hunk 6]
```diff
@@ -1775,7 +1775,7 @@ struct Slice(size_t _N, _Range)
     }
 
     /++
-    $(BLUE Partially or fully defined slice).
+    Partially or fully defined slice.
```
Comment (9il) [inline std/experimental/ndslice/slice.d]: Bold

File: std/experimental/ndslice/selection.d [hunk 7]
```diff
@@ -69,7 +69,7 @@ Params:
     K = sizes of dimension packs
 Returns:
     `pack!K` returns `Slice!(N-K, Slice!(K+1, Range))`;
-    `slice.pack!(K1, K2, ..., Kn)` is the same as `slice.pacKed!K1.pacKed!K2. ... pacKed!Kn`.
+    `slice.pack!(K1, K2, ..., Kn)` is the same as `slice.pack!K1.pack!K2. ... pack!Kn`.
```
Comment (9il) [inline std/experimental/ndslice/selection.d]: Wow!

File: std/experimental/ndslice/iteration.d [hunk 8]
```diff
@@ -9,24 +9,31 @@ All operators return slice of the same type as the type of the argument.
...
 )
 See also $(SUBREF selection, evertPack).
 
 $(BOOKTABLE $(H2 Iteration operators),
 
 $(TR $(TH Function Name) $(TH Description))
-$(T2 strided, `iotaSlice(13, 40).strided!(0, 1)(2, 5).shape` equals to `[7, 8]`.)
-$(T2 reversed, `slice.reversed!0` returns the slice with reversed direction of iteration for top level dimension.)
-$(T2 allReversed, `iotaSlice(4, 5).allReversed` equals to `20.iota.retro.sliced(4, 5)`.)
+$(T2 strided, Multiplies the stride of of a selected dimension by a factor.$(BR)
```
Comment (9il) [inline std/experimental/ndslice/iteration.d]: double `of`

**Comment by 9il** [comment]
probably `_` cause the error

**Comment by wilzbach** [comment]
> probably _ cause the error

Yes, but now I have no idea how to remove the boldness from the unittest description, see e.g.

http://dlang.org/phobos/std_experimental_ndslice_selection.html#.diagonal

(addressed all other comments)

**Comment by 9il** [comment]
Auto-merge toggled on

**Pull request merged**